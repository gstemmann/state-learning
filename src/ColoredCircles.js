import React, { useState } from "react";
import Circle from "./Circle";
import ButtonColor from "./ButtonColor";

function getRandom(min = 0, max = 100) {
    return Math.random() * (max - min) + min;
}

const ColoredCircles = () => {
    const [currentCircle, changeCircles] = useState([]);
    const addCircle = (newColor) => {
        changeCircles([...currentCircle, { color: newColor, x:getRandom(), y:getRandom() }]);
    }
    // const changePosition = (index) => {
    //     const copyOfCurrentCircle = [...currentCircle];
    //     copyOfCurrentCircle[index].x = getRandom();
    //     copyOfCurrentCircle[index].y = getRandom();
    //     return copyOfCurrentCircle;
    // }
    const changePosition = (index) => {
        changeCircles(currentCircle.map((circle, i) => {
            if (i === index) {
                return { ...circle, x: getRandom(), y: getRandom() };
            }
            return circle;
        }
        ));
    }
    return (
        <div className="ColoredCircles">
            <ButtonColor addCircle={addCircle}  options={['cornflowerblue', 'yellow']}/>
            {currentCircle.map(({ color, x, y }, index) => (
                <Circle changePosition={changePosition} 
                        color={color} 
                        index={index} 
                        key={index} 
                        x={x} 
                        y={y}/>  
            ))}
        </div>
    )
}

export default ColoredCircles;