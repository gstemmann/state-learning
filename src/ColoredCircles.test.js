import React from "react";
import { render, screen } from "@testing-library/react";
import ColoredCircles from "./ColoredCircles";

it('renders the ColoredCircles component', () => {
    render(<ColoredCircles />);
  });
  
  // snapshot test
  it("matches snapshot", function() {
      const {asFragment} = render(<ColoredCircles/>);
      expect(asFragment()).toMatchSnapshot();
    });
  
