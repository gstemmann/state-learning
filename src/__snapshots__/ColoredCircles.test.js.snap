// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColoredCircles"
  >
    <div
      class="ButtonColor"
    >
      <button>
        cornflowerblue
      </button>
      <button>
        yellow
      </button>
    </div>
  </div>
</DocumentFragment>
`;
